goaviatrix/controller_private_oob.go

```diff
@@ -0,0 +1,103 @@
+package goaviatrix
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+type PrivateOobResp struct {
+	Return  bool   `json:"return"`
+	Results string `json:"results"`
+	Reason  string `json:"reason"`
+}
+
+func (c *Client) EnablePrivateOob() error {
```

Comment:
Can use PostAPI with a custom CheckFunc instead. For example:
```golang
func (c *Client) UpdateAwsGuardDutyExcludedIPs(account *AwsGuardDutyAccount) error {
	data := map[string]string{
		"action":       "update_aws_guard_duty_excluded_ips",
		"CID":          c.CID,
		"account_name": account.AccountName,
		"region":       account.Region,
		"excluded_ips": strings.Join(account.ExcludedIPs, ","),
	}
	checkFunc := func(action, reason string, ret bool) error {
		if !ret && !strings.HasPrefix(reason, "No change in the exclude ip list for account") {
			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
		}
		return nil
	}
	return c.PostAPI(data["action"], data, checkFunc)
}
```

Reply:
Looks like this "checkFunc" can not cover the err of decoding?

Reply:
The decoding errors are caught by default.

The checkFunc is for once we decoded successfully and we decide on the error based on the the "Return" and "Reason"